main/src/library/MutList.flix

```diff
@@ -174,39 +447,103 @@ namespace MutList {
     ///
     /// If the given index `i` exceeds the length of the mutable list, no element is removed.
     ///
-    pub def remove(v: mut MutList[a], i: Int32): Unit = ???
+    pub def remove(i: Int32, v: mut MutList[a]): Unit & Impure =
+        let MutList(a, l) = v;
+        let len = deref l;
+        if (i < len) {
+            a := Array.init(j -> if (j < i) (deref a)[i] else (deref a)[i + 1], len - 1);
+            l := len - 1
+        }
+        else
+            ()
```

Comment:
If the number of elements in the MutList goes below 1/3 of the size of the backing array then you should shrink the array to 1/2 its size. (At least that is how I remember it).

The reason for doubling and shrinking is to ensure that /on average/ pop/push are O(1) amortized. I believe CLRS has some more information. 

The reason we don't immediately cut the array to 1/2 size when we can is to avoid the situation when you pop (reduce), push (double), pop (reduce) costing linear time for each.

Reply:
I had this in mind but failed to find any literature supporting it (I did not check in CLRS, though...).
I'll check CLRS and implement the shrinking logic.